lib/iris/cube.py

```diff
@@ -2205,12 +2205,11 @@ def new_cell_measure_dims(cm_):
             first_slice = next(slice_gen)
         except StopIteration:
             first_slice = None
-        if self._numpy_array is not None:
-            cube_data = self._numpy_array
-        elif self._dask_array is not None:
+
+        if self.has_lazy_data():
             cube_data = self._dask_array
         else:
-            raise ValueError('This cube has no data, slicing is not supported')
```

Comment:
This can never be hit. You have to supply a `data` variable to a cube constructor, and even if the data variable is not an array and not a dask object, whatever it _is_ still gets passed through `np.asarray()` and assigned to `cube._numpy_array`.